### DISCUSSION TITLE
Image control and large textures

### DISCUSSION THREAD

**Discussion by onslauth** [microsoft/WindowsAppSDK#4448] [Q&A]
I am using an `Image` control to display images. The images are loaded from memory into a `SoftwareBitmap` then converted to a `SoftwareBitmapSource` to set the `Image.Source` property.

Recently when testing a Surface Pro device, I ran into the following error message:
`Cannot create CanvasImageSource sized 19532 x 40; MaximumBitmapSizeInPixels for this device is 16384.`

I would like to know what would be the correct way to display large textures using the Image control?

**Comment by jeffstall** [comment]
Please take a look at [CompositionVirtualDrawingSurface](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.composition.compositionvirtualdrawingsurface?view=windows-app-sdk-1.5.) to handle very large sizes. It's a lower-level rendering/composition primitive, and the app will need to decide the image into tiles. There are hardware limits about the maximum size of a D3D texture.

**Comment by castorix** [comment]
> when testing a Surface Pro device,

On Windows 10, even without real graphic card, I did some tests and I can load a lot larger images (like 20000x3400)
(and _MaximumBitmapSizeInPixels_ is from Win2D; are you sure you don't use any other classes than just a SoftwareBitmap ?)
